ports/servoshell/app.rs

```diff
@@ -208,16 +216,36 @@ impl App {
             // Handle the event
             let mut consumed = false;
             if let Some(mut minibrowser) = app.minibrowser() {
-                if let winit::event::Event::WindowEvent { ref event, .. } = e {
-                    let response = minibrowser.context.on_event(&event);
-                    if response.repaint {
-                        // Request a winit redraw event, so we can recomposite, update and paint the
-                        // minibrowser, and present the new frame.
-                        window.winit_window().unwrap().request_redraw();
-                    }
-
-                    // TODO how do we handle the tab key? (see doc for consumed)
-                    consumed = response.consumed;
+                match e {
+                    winit::event::Event::WindowEvent {
+                        event: WindowEvent::ScaleFactorChanged { scale_factor, .. },
+                        ..
+                    } => {
+                        // Catch any ScaleFactorChanged events and set egui’s scale factor manually,
+                        // because we have our own logic for calculating the scale factor to use.
+                        let effective_scale_factor = window.hidpi_factor().get();
+                        info!(
+                            "window scale factor changed to {}, setting scale factor to {}",
+                            scale_factor, effective_scale_factor
+                        );
+                        minibrowser
+                            .context
+                            .egui_ctx
```

Comment:
Is there a reason that this isn't part of `minibrowser.context.on_event(...)`?  Is it just that this kind of event doesn't return a response? Perhaps returning an Option<...> in that case makes sense.

Reply:
Ah, I had forgotten that `egui_glow.rs` is essentially vendored code.